src/providers/pkcs11_provider/key_metadata.rs

```diff
@@ -0,0 +1,115 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use super::{KeyInfo, KeyMetadata, Pkcs11Provider};
+use crate::key_info_managers;
+use crate::key_info_managers::KeyTriple;
+use log::{error, warn};
+use parsec_interface::operations::psa_key_attributes::*;
+use parsec_interface::requests::{ResponseStatus, Result};
+
+impl Pkcs11Provider {
+    /// Gets a key identifier and key attributes from the Key Info Manager.
+    pub(super) fn get_key_info(&self, key_triple: &KeyTriple) -> Result<([u8; 4], Attributes)> {
+        let store_handle = self.key_info_store.read().expect("Local ID lock poisoned");
+        let key_metadata_cache = self
```

Comment:
Actually, I have just checked the `OnDiskKeyInfoManager` [implementation](https://github.com/parallaxsecond/parsec/blob/3ea3461da05bad9006572c306ccc28373d685ee9/src/key_info_managers/on_disk_manager/mod.rs#L34) and it already stores a cache of value it uses when getting them/checking if they exist. Which means that the cache might not be needed here 😬

Reply:
Lol, I didn't even check, I just assumed there was no cache.